config/default/kustomization.yaml

```diff
@@ -4,78 +4,15 @@
 apiVersion: kustomize.config.k8s.io/v1beta1
 kind: Kustomization
 
-# Adds namespace to all resources.
-namespace: application-system
-
-# Value of this field is prepended to the
-# names of all resources, e.g. a deployment named
-# "wordpress" becomes "alices-wordpress".
-# Note that it should also match with the prefix (text before '-') of the namespace
-# field above.
-namePrefix: kube-app-manager-
-
-# Labels to add to all resources and selectors.
 commonLabels:
   control-plane: kube-app-manager
+  app.kubernetes.io/name: kube-app-manager
+  app.kubernetes.io/version: dev
```

Comment:
Please don't add `app.kubernetes.io/version: dev` as a commonLabel. The commonLabel added here will be injected to the labelSelector of the StatefulSet for the manager. It will cause trouble when you update the manager to a different version.

Suggest to add it in the commonAnnotation.

Reply:
Good catch, thank you !
 
`app.kubernetes.io/version` is  one of the 5-6 recommended label as suggested by the kubernetes.io documentation.
I’m removing it, but I think it could also be a potential improvement of kustomize: not all labels are there for label selection, some are used for filtering/indexing/operator management.
I’ll open an issue on kustomize repo to followup on this.